registry-requirements.md

```diff
@@ -122,3 +69,48 @@ Same content as existing binding
+  - Reasons Behind the Requirement:
+    - Instead of WG learning each new protocol and media type, it is more efficient for people with a good understanding of the protocol or media type to write a binding.
+    - Engaging other communities.
+- A binding SHOULD correspond to specific TD specification version(s).
+  - Reason Behind the Requirement: A binding may not fit newer or older versions of a TD specification (e.g., `readproperty` can become `readprop`, or a new operation can arrive). Thus, at the time of writing a binding, it needs to be associated with one or more known TD specification versions.
+- The binding registry MUST be a separate document but associated with a TD version.
+  - Reasons Behind the Requirement: It is easier to update in the long term.
+- Association of a binding with the TD specification (registry entry) SHOULD be confirmed by the WoT Working Group. In other words, a person needs some permission and/or confirmation to authoritatively say that a given binding can be used with TD version X. The custodian of this registry is the WoT WG.
+  - Reasons Behind the Requirement: WoT WG is the manager of the registry.
+- If the WoT WG no longer exists, the W3C Team or their delegated entity becomes the custodian.
+  - Reasons Behind the Requirement: It should be possible to maintain the registry without the WoT WG.
+- It SHOULD be possible to register, update, or remove a registry entry for associating a binding to a specific TD specification outside of the REC lifecycle of the given TD specification.
+  - Reasons Behind the Requirement:
+    - Allowing new protocols to be bound to any TD version. 
+    - The process should outlive the WG if the WG ever comes to stop functioning or if the WG decides to not work on a TD specification version anymore.
+- There SHOULD NOT be two bindings for the same protocol, media type etc.
+  - Reasons Behind the Requirement:
+    - It breaks interoperability otherwise
+    - We should avoid confusing users, i.e. which binding should I use?
+    - It would complicate implementations
+  - DISCUSS: Clarify what happens when two ecosystems like OCF and LwM2M both use CoAP binding. The initial thinking is to register them as separate entries but clarify what they use from CoAP binding. A layered registry can be thought of as similar to language tags with `en` extending to `en-us` and `en-uk`, where the tags and entries are different but the association with `en` is expressed in the id. In our case, this would be `coap-ocf` and `coap-lwm2m`. A higher-level binding SHOULD NOT override or conflict with a lower-level binding, and this should be verified by the custodian, e.g., `cov:method` in CoAP binding should not be turned into `cov:operation` in the higher-level binding. The namespace (prefix and its values) defined in a binding CANNOT be redefined in any other binding.
```

Comment:
```suggestion
  - DISCUSS: Clarify what happens when two ecosystems like OCF and LwM2M both use CoAP binding. The initial thinking is to register them as separate entries and clarify what they use from CoAP binding. A layered registry can be thought of as similar to language tags with `en` extending to `en-us` and `en-uk`, where the tags and entries are different but the association with `en` is expressed in the ID. In our case, this would be `coap-ocf` and `coap-lwm2m`. A higher-level binding SHOULD NOT override or conflict with a lower-level binding, and this should be verified by the custodian, e.g., `cov:method` in CoAP binding should not be turned into `cov:operation` in the higher-level binding. The namespace (prefix and its values) defined in a binding CANNOT be redefined in any other binding.
```